tests/unit-tests/did_doc_metadata_test.go

```diff
@@ -1,88 +1,53 @@
-	validDid := "did:cheqd:mainnet:" + validIdentifier
-	validResourceId := "a09abea0-22e0-4b35-8f70-9cc3a6d0b5fd"
-	resourceData := []byte("test_checksum")
-	h := sha256.New()
-	h.Write(resourceData)
-	resourceMetadata := resourceTypes.Metadata{
-		CollectionId: validIdentifier,
-		Id:           validResourceId,
-		Name:         "Existing Resource Name",
-		ResourceType: "CL-Schema",
-		MediaType:    "application/json",
-		Checksum:     fmt.Sprintf("%x", h.Sum(nil)),
-	}
+var _ = Describe("Test NewResolutionDIDDocMetadata function", func() {
+	It("can resolution metadata with resource", func() {
+		metadata := &didTypes.Metadata{
+			VersionId:   ValidVersionId,
+			Deactivated: false,
+		}
 
-	created := resourceMetadata.Created.AsTime()
-	validMetadataResource := types.DereferencedResource{
-		ResourceURI:       validDid + types.RESOURCE_PATH + resourceMetadata.Id,
-		CollectionId:      resourceMetadata.CollectionId,
-		ResourceId:        resourceMetadata.Id,
-		Name:              resourceMetadata.Name,
-		ResourceType:      resourceMetadata.ResourceType,
-		MediaType:         resourceMetadata.MediaType,
-		Created:           &created,
-		Checksum:          resourceMetadata.Checksum,
-		PreviousVersionId: nil,
-		NextVersionId:     nil,
-	}
+		resources := []*resourceTypes.Metadata{
+			&ResourceMetadata,
+		}
 
-	subtests := []struct {
-		name           string
-		metadata       *didTypes.Metadata
-		resources      []*resourceTypes.Metadata
-		expectedResult types.ResolutionDidDocMetadata
-	}{
-		{
-			name: "metadata with resource",
-			metadata: &didTypes.Metadata{
-				VersionId:   "test_version_id",
-				Deactivated: false,
-			},
-			resources: []*resourceTypes.Metadata{&resourceMetadata},
-			expectedResult: types.ResolutionDidDocMetadata{
-				Created:     &EmptyTime,
-				Updated:     &EmptyTime,
-				Deactivated: false,
-				VersionId:   "test_version_id",
-				Resources:   []types.DereferencedResource{validMetadataResource},
-			},
-		},
-		{
-			name: "metadata without resources",
-			metadata: &didTypes.Metadata{
-				VersionId:   "test_version_id",
-				Deactivated: false,
-			},
-			expectedResult: types.ResolutionDidDocMetadata{
-				Created:     &EmptyTime,
-				Updated:     &EmptyTime,
-				VersionId:   "test_version_id",
-				Deactivated: false,
-			},
-		},
-	}
+		expectedResult := types.ResolutionDidDocMetadata{
+			Created:     nil,
+			Updated:     nil,
+			Deactivated: false,
+			VersionId:   ValidVersionId,
+			Resources:   []types.DereferencedResource{ValidMetadataResource},
+		}
 
-	for _, subtest := range subtests {
-		t.Run(subtest.name, func(t *testing.T) {
-			result := types.NewResolutionDidDocMetadata(validDid, subtest.metadata, subtest.resources)
+		result := types.NewResolutionDidDocMetadata(ValidDid, metadata, resources)
+		Expect(result).To(Equal(expectedResult))
+	})
 
-			require.EqualValues(t, subtest.expectedResult, result)
-		})
-	}
-}
+	It("can resolution metadata without resource", func() {
```

Comment:
Here it may be a sentence to understand the actions inside. Let's make it more human readable, like "It can create the structure without resource" or something like this.

Reply:
Done! ✅